File: Python/count_the_divisors.py
#  SOLUTION ONE
def divisors(n):
    c = 1
    for d in range(1, n):
        if n % d == 0:
            c += 1
    return c

#  SOLUTION TWO


def divisors(n):
    return len([l_div for l_div in range(1, n + 1) if n % l_div == 0])


print(divisors(4))
print(divisors(5))
print(divisors(12))
print(divisors(30))

File: Ruby/count_the_divisors.rb
# SOULTION ONE
def divisors(n)
  i = 1 
  count = 0
  while i <= n
    if n % i == 0
      count += 1
    end
  i += 1
  end
  return count
end

# SOLUTION TWO
def divisors(n)
  return (1..n).count {|d| n % d == 0}
end

p divisors(4)
p divisors(5)
p divisors(12)
p divisors(30)